library/std/build.rs

```diff
@@ -3,66 +3,23 @@ use std::env;
 fn main() {
     println!("cargo:rerun-if-changed=build.rs");
     let target = env::var("TARGET").expect("TARGET was not set");
-    if target.contains("linux") {
-        if target.contains("android") {
-            println!("cargo:rustc-link-lib=dl");
-            println!("cargo:rustc-link-lib=log");
-            println!("cargo:rustc-link-lib=gcc");
-        }
-    } else if target.contains("freebsd") {
-        println!("cargo:rustc-link-lib=execinfo");
-        println!("cargo:rustc-link-lib=pthread");
+    if target.contains("freebsd") {
         if env::var("RUST_STD_FREEBSD_12_ABI").is_ok() {
             println!("cargo:rustc-cfg=freebsd12");
         }
```

Comment:
You could get rid of this too with [`option_env!`](https://doc.rust-lang.org/std/macro.option_env.html), right? Not necessarily here, but it would be possible?

And below you could get rid of

```rust
    println!("cargo:rustc-env=STD_ENV_ARCH={}", env::var("CARGO_CFG_TARGET_ARCH").unwrap());
    println!("cargo:rustc-cfg=backtrace_in_libstd");
```
by using `env!` and unconditionally setting `backtrace_in_libstd` in rustbuild?

Reply:
Just going to note that `{option_}env!` runs at compile-time, not runtime. Though is that effectively the same for build scripts?

Reply:
Runtime for a build script is the same as compile time for the main code, yes.

Reply:
But are environment variables at compile-time equal to environment variables at runtime in a build script?

Reply:
`CARGO_CFG_TARGET_ARCH` is not set when running rustc. `RUST_STD_FREEBSD_12_ABI` is used to set the `freebsd12` cfg, which is used to conditionally compile some code. There is no way to conditionally compile some code based on env vars.

Reply:
> and unconditionally setting backtrace_in_libstd in rustbuild?

That would require a change to cargo for -Zbuild-std, and a change to xargo.